eth2util/keystore/keystore_internal_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package keystore
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_orderByKeystoreNum(t *testing.T) {
+	tests := []struct {
+		name     string
+		files    []string
+		want     []string
+		errCheck require.ErrorAssertionFunc
+	}{
+		{
+			"keystore secure files",
+			[]string{
+				"/keystore-10.json",
+				"/keystore-1.json",
+				"/keystore-3.json",
+				"/keystore-2.json",
+			},
+			[]string{
+				"/keystore-1.json",
+				"/keystore-2.json",
+				"/keystore-3.json",
+				"/keystore-10.json",
+			},
+			require.NoError,
+		},
+		{
+			"keystore insecure files",
+			[]string{
+				"/keystore-insecure-10.json",
+				"/keystore-insecure-1.json",
+				"/keystore-insecure-3.json",
+				"/keystore-insecure-2.json",
+			},
+			[]string{
+				"/keystore-insecure-1.json",
+				"/keystore-insecure-2.json",
+				"/keystore-insecure-3.json",
+				"/keystore-insecure-10.json",
+			},
+			require.NoError,
+		},
+		{
+			"filenames that do not pass regex error early",
+			[]string{
+				"/keystore-insecure-fail.json",
+				"/keystore-insecure-failtoo.json",
+			},
+			nil,
+			require.Error,
+		},
+		{
+			"single file path yields the exact same thing",
+			[]string{
+				"/keystore-0.json",
```

Comment:
add a test for missing 0 index 🙏

Reply:
Already tested in `TestStoreLoadKeysAllNonSequentialIdx`.